hooks/open-telemetry/pkg/otel.go

```diff
@@ -2,165 +2,60 @@ package otel
 
 import (
 	"context"
-	"encoding/json"
-	"errors"
-	"fmt"
-	"sync"
 
 	of "github.com/open-feature/go-sdk/pkg/openfeature"
 	"go.opentelemetry.io/otel/attribute"
-	"go.opentelemetry.io/otel/codes"
 	"go.opentelemetry.io/otel/trace"
 )
 
 const (
-	// AttributeFlagKey trace attribute name for the evaluated flag key
-	AttributeFlagKey = "feature_flag.flag_key"
-	// AttributeProviderName trace attribute name for the provider used for evaluation
-	AttributeProviderName = "feature_flag.provider_name"
-	// AttributeEvaluatedVariant trace attribute name for the variant returned from the flag evaluation
-	AttributeEvaluatedVariant = "feature_flag.evaluated_variant"
-	// AttributeEvaluatedValue trace attribute name for the value returned from the flag evaluation
-	AttributeEvaluatedValue = "feature_flag.evaluated_value"
-	// AttributeTraceName the name of the trace exported by the hook
-	AttributeTraceName = "github.com/open-feature/go-sdk-contrib/hooks/open-telemetry/pkg"
+	EventName                 = "feature_flag"
+	EventPropertyFlagKey      = "feature_flag.key"
+	EventPropertyProviderName = "feature_flag.provider_name"
+	EventPropertyVariant      = "feature_flag.variant"
 )
 
 type Hook struct {
-	spans        map[string]*mutexWrapper
-	ctx          context.Context
-	wg           *sync.WaitGroup
-	tracerClient tracerClientInterface
-}
-
-// mutex wrapper is used to prevent colliding keys from overwriting each other / closing a partially completed span
-type mutexWrapper struct {
-	mu *sync.Mutex
-	ss *storedSpan // ss has been set as a pointer so it can be set to nil during cleanup
-}
-type storedSpan struct {
-	cancel func()
-	span   trace.Span
+	ctx context.Context
 }
 
 // NewHook return a reference to a new instance of the OpenTelemetry Hook
-func NewHook() *Hook {
+func NewHook(ctx context.Context) *Hook {
 	return &Hook{
-		tracerClient: &tracerClient{},
+		ctx: ctx,
 	}
 }
 
-// Wait blocks until all spans have been closed
-func (h *Hook) Wait() {
-	h.wg.Wait()
-}
-
-// WithContext sets the parent context used when new spans are formed. When the parent context is closed so are all spans.
-// if no parent context is provided context.Background() is used internally
-func (h *Hook) WithContext(ctx context.Context) {
-	h.ctx = ctx
-}
-
-// Before creates the flag evaluations open-telemetry span and sets the FlagKey and ProviderName attributes
+// Before method is unused for this hook
```

Comment:
Could include `UnimplementedHook` from the go-sdk in your Hook struct to avoid implementing empty funcs

Reply:
updated